base/server/python/pki/server/cli/cert.py

```diff
@@ -959,11 +959,17 @@ def execute(self, argv):
                 logger.info('Exporting %s certificate into %s.', cert_id, cert_file)
 
                 cert_data = cert.get('data')
-                if cert_data is None:
-                    logger.error('Unable to find certificate data for %s', cert_id)
-                    sys.exit(1)
+                if cert_data:
+                    cert_data = pki.nssdb.convert_cert(cert_data, 'base64', 'pem')
+                else:
+                    crt_path = os.path.join(instance.conf_dir, 'conf', 'certs', f'{cert_id}.crt')
```

Comment:
I think we don't need the `conf` part since `instance.conf_dir` is already pointing to the config folder.

Reply:
Unfortunately I think we cannot use f-string because the code needs to be able to run on RHEL 8 which only has Python 3.6.

Reply:
> Unfortunately I think we cannot use f-string because the code needs to be able to run on RHEL 8 which only has Python 3.6.

f-string are available from 3.6 (https://peps.python.org/pep-0498) and in my test VM they works but I have modified with '+' concatenation to avoid compatibility issue.

Reply:
About f-string, looks like you're right. I was comparing Python 3.6 and 3.7 docs, apparently the f-string is described in different sections:
https://docs.python.org/3.7/tutorial/inputoutput.html
https://docs.python.org/3.6/tutorial/inputoutput.html

Reply:
Generally a Tomcat instance has a `conf` folder under the instance folder that contains Tomcat config files (e.g. `server.xml`): https://tomcat.apache.org/tomcat-9.0-doc/introduction.html

Following the same convention, in a PKI the folder would be `/var/lib/pki/<instance name>/conf`. However, Fedora requires that config files to be stored under `/etc`, so the `/var/lib/pki/<instance name>/conf` is actually a link to `/etc/pki/pki-tomcat`, but here we don't need to add `conf` in the path anymore since everything under `/etc` are config files, so it would be redundant.

Yeah, we can certainly add something like `certs_dir` to define the location for system certs and CSRs.

Reply:
Reading the issue you wrote the certificate should go in `<instance>/conf/certs` folder and I was thinking that `<instance>` is `/etc/pki/<instance_name>` but I was wrong, that is the conf folder if IIUC.

I will fix this directly in master. Thanks for the clarification.

Reply:
Thanks!